ACE/ace/os_include/os_dirent.h

```diff
@@ -46,7 +46,7 @@ struct dirent {
 };
 #endif /* ACE_HAS_DIRENT */
 
-#if defined (ACE_LACKS_STRUCT_DIR)
+#if defined (ACE_LACKS_DIRENT_H)
```

Comment:
This change doesn't look ok, we have a separate define to check if struct dirent is missing, looks better to update ACE_LACKS_STRUCT_DIR to ACE_LACKS_STRUCT_DIRENT